v0/destinations/mp/transform.js

```diff
@@ -1,101 +1,141 @@
 const get = require("get-value");
-const set = require("set-value");
 const { EventType } = require("../../../constants");
 const {
   removeUndefinedValues,
   defaultRequestConfig,
   defaultPostRequestConfig,
   constructPayload,
-  isDefined,
   getBrowserInfo,
   getValuesAsArrayFromConfig,
   toUnixTimestamp,
   getTimeDifference,
   getErrorRespEvents,
   getSuccessRespEvents,
   CustomError,
-  isAppleFamily,
-  getFullName,
-  extractCustomFields
+  isHttpStatusSuccess
 } = require("../../util");
+const { ConfigCategory, mappingConfig } = require("./config");
+
+const { httpPOST } = require("../../../adapters/network");
+
 const {
-  ConfigCategory,
-  mappingConfig,
-  MP_IDENTIFY_EXCLUSION_LIST,
-  GEO_SOURCE_ALLOWED_VALUES
-} = require("./config");
-
-// ref: https://help.mixpanel.com/hc/en-us/articles/115004613766-Default-Properties-Collected-by-Mixpanel 
-const mPIdentifyConfigJson = mappingConfig[ConfigCategory.IDENTIFY.name];
-const mPProfileAndroidConfigJson =
-  mappingConfig[ConfigCategory.PROFILE_ANDROID.name];
-const mPProfileIosConfigJson = mappingConfig[ConfigCategory.PROFILE_IOS.name];
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+
+const { createIdentifyResponse } = require("./util");
+
+// ref: https://help.mixpanel.com/hc/en-us/articles/115004613766-Default-Properties-Collected-by-Mixpanel
 const mPEventPropertiesConfigJson =
   mappingConfig[ConfigCategory.EVENT_PROPERTIES.name];
 
 function getEventTime(message) {
   return new Date(message.timestamp).toISOString();
 }
 
-function responseBuilderSimple(parameters, message, eventType, destConfig) {
-  let headers = {};
-  let endpoint =
-    destConfig.dataResidency === "eu"
-      ? "https://api-eu.mixpanel.com/engage/"
-      : "https://api.mixpanel.com/engage/";
-
-  if (
-    eventType !== EventType.IDENTIFY &&
-    eventType !== EventType.GROUP &&
-    eventType !== "revenue"
-  ) {
-    const duration = getTimeDifference(message.timestamp);
-    if (duration.days <= 5) {
-      endpoint =
-        destConfig.dataResidency === "eu"
-          ? "https://api-eu.mixpanel.com/track/"
-          : "https://api.mixpanel.com/track/";
-    } else if (duration.years > 5) {
-      throw new CustomError(
-        "Event timestamp should be within last 5 years",
-        400
-      );
-    } else {
-      endpoint =
-        destConfig.dataResidency === "eu"
-          ? "https://api-eu.mixpanel.com/import/"
-          : "https://api.mixpanel.com/import/";
-      if (destConfig.apiSecret) {
-        headers = {
-          Authorization: `Basic ${Buffer.from(
-            `${destConfig.apiSecret}:`
-          ).toString("base64")}`
-        };
-      } else {
-        throw new CustomError(
-          "Event timestamp is older than 5 days and no apisecret is provided in destination config.",
-          400
-        );
-      }
-    }
+async function createUser(response) {
```

Comment:
Better add documentation for this function

Reply:
Updated